src/Deanery.java

```diff
@@ -0,0 +1,111 @@
+                "\n" +
+                "------------------------------------------";
+    }
+
+    public static void serve(Student student, String roomNumber, int hour) {
+        if (!(roomNumber.equals(Rooms.A123.toString())||roomNumber.equals(Rooms.B123.toString())||roomNumber.equals(Rooms.C124.toString())||roomNumber.equals(Rooms.D124.toString()))) {
+            throw new RuntimeException("The room does not match any od Deanery's rooms");
+        }
+        if (Rooms.valueOf(roomNumber) == Rooms.A123) {
+            if (student.getTypeOfStudy() == "FULL_TIME") {
+                if (hour >= Rooms.A123.getRoomHour().getStartHour() && hour <= Rooms.A123.getRoomHour().getEndHour()) {
+                    System.out.println(createRaport(student, roomNumber));
+                }
+                else {
+                    throw new RuntimeException("Wrong hours to serve student!");
+                }
+            }
+            else {
+                throw new RuntimeException("Student is on a part-time type of study!");
+            }
+        } else if (Rooms.valueOf(roomNumber) == Rooms.B123) {
+            if (student.getTypeOfStudy() == "FULL_TIME") {
+                if (hour >= Rooms.B123.getRoomHour().getStartHour() && hour <= Rooms.B123.getRoomHour().getEndHour()) {
+                    System.out.println(createRaport(student, roomNumber));
+                } else {
+                    throw new RuntimeException("Wrong hours to serve student!");
+                }
+            }
+            else {
+                throw new RuntimeException("Student is on a part-time type of study!");
+            }
+        } else if (Rooms.valueOf(roomNumber) == Rooms.C124) {
+            if (student.getTypeOfStudy() == "PART_TIME") {
+                if (hour >= Rooms.C124.getRoomHour().getStartHour() && hour <= Rooms.C124.getRoomHour().getEndHour()) {
+                    System.out.println(createRaport(student, roomNumber));
+                }
+                else {
+                    throw new RuntimeException("Wrong hours to serve student!");
+                }
+            }
+            else {
+                throw new RuntimeException("Student is on a full-time type of study!");
+            }
+
+        } else if (Rooms.valueOf(roomNumber) == Rooms.D124) {
+            if(student.getTypeOfStudy() == "PART_TIME") {
+                if (hour >= Rooms.D124.getRoomHour().getStartHour() && hour <= Rooms.D124.getRoomHour().getEndHour()) {
+                    System.out.println(createRaport(student, roomNumber));
+                }
+                else {
+                    throw new RuntimeException("Wrong hours to serve student!");
+                }
+            }
+            else {
+                throw new RuntimeException("Student is on a full-time type of study!");
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+
+        ArrayList<Student> studentList = new ArrayList<>();
```

Comment:
One thing here -> maybe instead of using a list, it would make more sense to e.g. a map with student ID as a key? retrieving a student would be very fast and simple then -> no iteration would be required